app/controllers/donors_controller.rb

```diff
@@ -40,4 +41,20 @@ def find_donor(id)
   def donor_params
     params.require(:donor).permit(:name, :phone_number, :email_address, :address, donations: [:cause_id])
   end
+
+  def filtered_donors
+    if !params[:cause_id].blank?
+      Donor.cause_id(params[:cause_id])
```

Comment:
I tried that and also added the following code in the Cause model:

```ruby
has_many :donors, -> { uniq }, through: :donations
```

But I am getting the error:

PG::InvalidColumnReference: ERROR:  for SELECT DISTINCT, ORDER BY expressions must appear in select list
LINE 1: ...r_id" WHERE "donations"."cause_id" = $1  ORDER BY "donations...